node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeRebooter.java

```diff
@@ -49,13 +52,32 @@ protected void maintain() {
     }
     
     private boolean shouldReboot(Node node) {
-        var rebootEvents = EnumSet.of(History.Event.Type.rebooted, History.Event.Type.osUpgraded);
-        var acceptableRebootInstant = clock.instant().minus(rebootInterval);
+        if (node.status().reboot().pending()) return false;
+
+        var rebootEvents = EnumSet.of(History.Event.Type.provisioned, History.Event.Type.rebooted, History.Event.Type.osUpgraded);
+        var rebootInterval = Duration.ofDays(rebootIntervalInDays.value());
+
+        Optional<Duration> overdue = node.history().events().stream()
+                .filter(event -> rebootEvents.contains(event.type()))
+                .map(History.Event::at)
+                .max(Comparator.naturalOrder())
+                .map(lastReboot -> Duration.between(lastReboot, clock.instant()).minus(rebootInterval));
 
-        if (rebootEvents.stream().anyMatch(event -> node.history().hasEventAfter(event, acceptableRebootInstant)))
+        if (overdue.isEmpty()) // should never happen as all !docker-container should have provisioned timestamp
+            return random.nextDouble() < interval().getSeconds() / (double) rebootInterval.getSeconds();
+
+        if (overdue.get().isNegative())
             return false;
-        else // schedule with a probability such that reboots of nodes are spread roughly over the reboot interval
-            return random.nextDouble() < (double) interval().getSeconds() / (double)rebootInterval.getSeconds();
+
+        // Use a probability such that each maintain() schedules the same number of reboots,
+        // as long as 0 <= overdue <= rebootInterval, with the last maintain() in that interval
+        // naturally scheduling the remaining with probability 1.
+
+        int configServers = 3;
```

Comment:
Should be determined automatically by using `CuratorDatabaseClient#cluster`.

Reply:
Fixed in #11313